### PR TITLE
Disaster recovery feature via mongodb replica set

### PR DESCRIPTION
**Pull request by jkralik**
## Description
The primary and standby cluster MongoDB members are in the same MongoDB replica set. The standby cluster members are configured as [hidden](https://www.mongodb.com/docs/manual/core/replica-set-hidden-member), [delayed](https://www.mongodb.com/docs/manual/core/replica-set-delayed-member/), and with [zero priority](https://www.mongodb.com/docs/manual/core/replica-set-priority-0-member/). When the primary cluster goes down, the standby cluster MongoDB members are promoted to standby state—one of them will become primary by administrator. After the primary is back online, the primary cluster members will be demoted to hidden. For switching back, the primary cluster members will be promoted to secondary MongoDB members and standby cluster members will be demoted.

## Other changes
- Upgrade mongodb to v7

### CODE REVIEW

**Comment by jkralik** [comment]
@coderabbitai review

File: tools/mongodb/standby-tool/main.go [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+			return fmt.Errorf("failed to connect to MongoDB: %w", err2)
+		}
+		defer func() { _ = newPrimaryMemberClient.Disconnect(context.Background()) }()
+		err = app.setHiddenMembers(ctx, newPrimaryMemberClient, app.Config.ReplicaSet.ForceUpdate, newPrimaryMember, []string{primaryMember})
+		if err != nil {
+			return fmt.Errorf("failed to set hidden members: %w", err)
+		}
+	}
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (jkralik) [reply tools/mongodb/standby-tool/main.go]: create a suggestion

File: tools/mongodb/admin-tool/main.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	MongoURI              string
+	Timeout               time.Duration
+	DirectConnection      bool
+}
+
+func parseArgs() MongoArgs {
+	var args MongoArgs
+
+	flag.BoolVar(&args.TLS, "tls", false, "Enable TLS")
+	flag.BoolVar(&args.TLS, "directConnection", false, "Direct connection")
```
Comment (Danielius1922) [inline tools/mongodb/admin-tool/main.go]: `&args.TLS` -> `&args.DirectConnection`

File: charts/plgd-hub/templates/coap-gateway/client-crt.yaml [hunk 3]
```diff
@@ -1,25 +1,28 @@
-{{- if and .Values.mongodb.enabled .Values.certmanager.enabled }}
+{{- $clientCertName := include "plgd-hub.coapgateway.clientCertName" . }}
+{{- if and $clientCertName .Values.certmanager.enabled .Values.coapgateway.enabled }}
+{{- $serviceDns := include "plgd-hub.coapgateway.fullname" . }}
```
Comment (Danielius1922) [inline charts/plgd-hub/templates/coap-gateway/client-crt.yaml]: `$serviceDns` doesn't seem to be used anywhere

File: charts/plgd-hub/values.yaml [hunk 4]
```diff
@@ -99,30 +217,42 @@ mongodb:
...
         secretName: mongodb-cm-crt
+    - name: mongodb-extra-ca-pool
+      secret:
+        secretName: mongodb-extra-ca-pool  
   initContainers:
     # workaround for https://github.com/bitnami/charts/issues/10264, https://jira.mongodb.org/browse/MONGOSH-1240
     - name: mongo-binary
-      image: docker.io/bitnami/mongodb:5.0.10-debian-11-r3
-      imagePullPolicy: "IfNotPresent"
+      image: ghcr.io/plgd-dev/hub/mongodb-admin-tool:vnext-pr1311
```
Comment (Danielius1922) [inline charts/plgd-hub/values.yaml]: this will be changed after merge?

File: charts/plgd-hub/values.yaml [hunk 5]
```diff
@@ -99,30 +217,42 @@ mongodb:
...
+      imagePullPolicy: "Always"
       volumeMounts:
         - name: mongodb-crt
           mountPath: "/certs"
         - name: mongodb-cm-crt
           mountPath: "/certs-original"
+        - name: mongodb-extra-ca-pool
+          mountPath: "/certs/extra/storage"
+      securityContext:
+        runAsUser: 1001
```
Comment (Danielius1922) [inline charts/plgd-hub/values.yaml:242]: is it guaranteed that user 1001 exists?

**Comment by Danielius1922** [comment]
@jkralik can you clean up SC issues, nothing important there, just some smells

File: charts/plgd-hub/values.yaml (hunk 4 above)
Comment (jkralik) [reply charts/plgd-hub/values.yaml]: before merge and yes :)

File: charts/plgd-hub/values.yaml (hunk 5 above)
Comment (jkralik) [reply charts/plgd-hub/values.yaml:242]: Yes it is used in mongodb 7.0

File: tools/mongodb/admin-tool/main.go (hunk 2 above)
Comment (jkralik) [reply tools/mongodb/admin-tool/main.go]: fixed

File: charts/plgd-hub/templates/coap-gateway/client-crt.yaml (hunk 3 above)
Comment (jkralik) [reply charts/plgd-hub/templates/coap-gateway/client-crt.yaml]: removed

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**